### PR TITLE
[coost] rename cocoyaxi and upgrade to coost v3.0.0

### PR DESCRIPTION
**Pull request by idealvin**
Rename cocoyaxi to coost, and upgrade to v3.0.0.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all / linux, windows, ...>, <Yes/No>

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by LilyWangLL** [comment]
Thanks for your PR. If you rename `cocoyaxi` to `coost`, you need also handle **portfile.cmake** and **vcpkg.json** of `coost`. You can refer the change of PR https://github.com/microsoft/vcpkg/pull/26633.

**Comment by idealvin** [comment]
> #26633

@LilyWangLL  Can we just remove cocoyaxi?

**Comment by LilyWangLL** [comment]
> > #26633
> 
> @LilyWangLL Can we just remove cocoyaxi? It will not be supported any more.

We can't just remove `cocoyaxi`, because someone may have installed it, remove this port will cause their project failure.
And adding dependency on old port can avoid the situation that the user cannot find the port without knowing this change.

**Comment by idealvin** [comment]
> > > #26633
> > 
> > 
> > @LilyWangLL Can we just remove cocoyaxi? It will not be supported any more.
> 
> We can't just remove `cocoyaxi`, because someone may have installed it, remove this port will cause their project failure. And adding dependency on old port can avoid the situation that the user cannot find the port without knowing this change.

Ok. I have fixed it now.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/coost/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "coost",
+  "version-semver": "3.0.0",
```
[the remark this answers is not in the document]
Comment (LilyWangLL) [reply ports/coost/vcpkg.json:3]: Sorry, I comment here, but I forget request changes. I click it just a moment ago.
Ping @idealvin

**Review by LilyWangLL — APPROVED** [review]

**Comment by vicroms** [comment]
We need a policy for package renames, this is the second time that this library has been renamed. And although I don't oppose the name change, we should make sure that the names we use for packages are stable for our users' sake.

**Comment by dan-shaw** [comment]
Closing this PR as there has been no progress in the last 6 months. Feel free to open a new PR if you are still working on this

**Pull request closed without merging**